romeo/src/state.rs

```diff
@@ -49,32 +58,29 @@ pub struct DepositInfo {
 }
 
 impl Deposit {
-    fn mint(&self) -> Option<Task> {
-        match self.mint.as_ref() {
-            Some(Response {
-                status: TransactionStatus::Broadcasted,
-                txid,
-            }) => Some(Task::CheckStacksTransactionStatus(*txid)),
-            // TODO: Think about removing deposits at this stage #99
-            Some(Response {
-                status: TransactionStatus::Confirmed,
-                ..
-            }) => None,
-            Some(Response {
-                status: TransactionStatus::Rejected,
-                ..
-            }) => panic!("Mint transaction rejected"),
-            // TODO: Confirm that the deposit wallet is correct
-            None => Some(Task::CreateMint(self.info.clone())),
+    fn request_work(&mut self) -> Option<Task> {
+        match self.mint {
+            None => {
+                self.mint = Some(TransactionRequest::Created);
+                Some(Task::CreateMint(self.info.clone()))
+            }
+            Some(TransactionRequest::Created)
+            | Some(TransactionRequest::Acknowledged(_, TransactionStatus::Confirmed)) => None,
+            Some(TransactionRequest::Acknowledged(txid, TransactionStatus::Broadcasted)) => {
+                Some(Task::CheckStacksTransactionStatus(txid))
+            }
+            Some(TransactionRequest::Acknowledged(txid, TransactionStatus::Rejected)) => {
+                panic!("Mint transaction rejected: {}", txid)
```

Comment:
Perhaps we should consider adding retry logic here as well. We should not crash on rejected transactions.

Reply:
At this point, retry has failed with a permanent error. I assume one reason for this is if the stacks node it is broadcasting to is just down. At this point, do we expect Romeo to log an error and retry? What other things would cause a permanent error? In those cases, would we expect Romeo to skip the transaction?